ExpenseTracker/operator/operator.go

```diff
@@ -1,219 +1,195 @@
+
+	if err != nil {
+		return nil, err
+	}
+	return conn, nil
+}
+
+// Checks that expense if provided id exists.
+func ExpenseExists(conn *pgx.Conn, id int) (bool, error) {
+	var exists bool
+	err := conn.QueryRow(context.Background(),
+		`SELECT EXISTS(SELECT 1 FROM public.Expenses WHERE id = $1)`, id).Scan(&exists)
+	return exists, err
+}
+
+// Adds a promoted expense (need to validate).
+func CreateExpense(conn *pgx.Conn, e Expense) error {
+	query := `INSERT INTO public.Expenses (Date, Description, Amount) VALUES (@Date, @Description, @Amount)`
+
+	args := pgx.NamedArgs{
+		"Date":        e.Date,
+		"Description": e.Desc,
+		"Amount":      e.Amount,
+	}
+
+	_, err := conn.Exec(context.Background(), query, args)
+	if err != nil {
+		return fmt.Errorf("error inserting expense: %w", err)
+	}
+	return nil
+}
+
+// Returns an array of all expenses available.
+func GetAllExpensesFromDb(conn *pgx.Conn) ([]Expense, error) {
+	query := `SELECT * FROM public.Expenses`
+
+	rows, err := conn.Query(context.Background(), query)
+	if err != nil {
+		fmt.Println("Error querying the table")
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	var expenses []Expense
+
+	for rows.Next() {
+		var exp Expense
+		err := rows.Scan(&exp.ID, &exp.Date, &exp.Desc, &exp.Amount)
+		if err != nil {
+			fmt.Println("Error fetching expense details")
+			return expenses, err
+		}
+		expenses = append(expenses, exp)
+	}
+	return expenses, nil
+}
+
+// Replaces expense with promoted id with new given expense.
+func UpdateExpense(conn *pgx.Conn, id int, exp Expense) error {
+	query := `
+		UPDATE public.Expenses
+		SET Date = @Date, Description = @Description, Amount = @Amount 
+		WHERE id = @id
+	`
+
+	args := pgx.NamedArgs{
+		"id":          id,
+		"Date":        exp.Date,
+		"Description": exp.Desc,
+		"Amount":      exp.Amount,
+	}
+
+	_, err := conn.Exec(context.Background(), query, args)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Deletes an expense with promoted id.
+func DeleteExpense(conn *pgx.Conn, id int) error {
+	query := `
+	DELETE FROM public.Expenses WHERE id = @id`
+
+	args := pgx.NamedArgs{
+		"id": id,
+	}
+
+	_, err := conn.Exec(context.Background(), query, args)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	return nil
+}
+
+// to help count sum of array.
+func sumOfArray(i []int) int {
+	summ := 0
+	for _, v := range i {
+		summ += v
+	}
+	return summ
+}
+
+// Returns an amount of all expenses.
+func Summary(conn *pgx.Conn) (int, error) {
+	query := `
+	SELECT e.Amount FROM public.Expenses as e
+	`
+
+	rows, err := conn.Query(context.Background(), query)
+
+	if err != nil {
+		fmt.Println("Error getting summary")
+		return 0, err
+	}
+	defer rows.Close()
+
+	var amounts []int
+
+	for rows.Next() {
+		var amount int
+		err := rows.Scan(&amount)
+		if err != nil {
+			fmt.Println("Error getting one amount")
+			return sumOfArray(amounts), err
```

Comment:
```suggestion
			return 0, err
```
Не стоит возвращать из функции какой-то частичный результат. Раз произошла ошибка, то мы не смогли посчитать сумму всех затрат. Это значит, что "всей суммы" мы не знаем. А значит, результат должен представлять собой пустое значение (в данном случае - `0`).

В целом в go есть конвенция, что если функция вернула ошибку, то результат этой функции использовать нельзя. Не стоит создавать ситуаций, когда при возникновении ошибки функция пытается вернуть какой-то частичный результат. Какой-то другой разработчик может заметить, что, несмотря на ошибку, функция возвращает результат, и попытается воспользоваться этим результатом. Например, попробует показать его пользователю. Но при этом разработчик не учтет, что результат неполный и (показывая его) он вводит пользователя в заблуждение.